packages/compose/src/hooks/use-constrained-tabbing/index.js

```diff
@@ -64,7 +64,7 @@ function useConstrainedTabbing() {
 			node[ domAction ]( trap );
 			trap.focus();
 			// Remove after the browser moves focus to the next element.
-			timeoutId = setTimeout( () => node.removeChild( trap ) );
+			timeoutId = setTimeout( () => node.removeChild( trap ), 10 );
```

Comment:
I have made the following changes and have confirmed that it works as expected:

```javascript
// Remove itself when the trap loses focus.
trap.addEventListener( 'blur', () => node.removeChild( trap ) );

trap.focus();
```

I couldn't figure out how to remove the event listener when the element itself is deleted, but is it also need to remove the event when the element is deleted?

Reply:
I'm not an expert but I think the element itself _and_ the event listener are retained in memory because there's still a reference to it. To test: dump the trap element to the console after it's removed from the DOM, see that it's still in memory::
```
			trap.addEventListener( 'blur', () => {
				node.removeChild( trap );
				console.log( trap );
			} );
```

Since a new `trap` element and event listener will be created at each Tab key press on the last and first tabbable elements, I suspect this could lead to some memory leak. Again, I'm not an expert but I think that declaring the `trap` element with `let` and then assigning it to `null` after it's removed could be an option e.g. `trap = null;`.
Would appreciate feedback from someone with better knowledge than me.

Reply:
> I couldn't figure out how to remove the event listener when the element itself is deleted, but is it also need to remove the event when the element is deleted?

It's my understanding that the listener does not need to be removed as long as there is no persisting reference to the element. It looks like in this case it'd be fine because each time the `onKeyDown` handler creates `trap`, the only closure around it is the blur listener itself. However, to be sure about it you could use the [`once`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#once) option when adding the listener:

```js
// Remove itself when the trap loses focus.
trap.addEventListener( 'blur', () => trap.remove(), { once: true } );
```
That example also switches to `remove` instead of `removeChild` for saving a few bits. Since IE11 doesn't need supported that should be safe.

Also I couldn't quickly find a outside source to back up my statements but there is this bit from the the jsdoc of `useRefEffect`: https://github.com/WordPress/gutenberg/blob/a12ad38cc0c2692f9c025c3d63e0a394a1e29461/packages/compose/src/hooks/use-ref-effect/index.ts#L20-L23
Which agrees and indicates that this hook itself could drop the cleanup function here:
https://github.com/WordPress/gutenberg/blob/a12ad38cc0c2692f9c025c3d63e0a394a1e29461/packages/compose/src/hooks/use-constrained-tabbing/index.js#L73-L76